lib/MongoDB/_URI.pm

```diff
@@ -141,12 +156,166 @@ sub _parse_doc {
     return $set;
 }
 
+use Devel::Dwarn;
+sub _parse_options {
+    my ( $self, $valid, $result, $err_unsupported ) = @_;
+
+    my %parsed;
+    for my $opt ( split '&', $result->{options} ) {
+        my @kv = split '=', $opt, -1;
+        MongoDB::UsageError->throw("expected key value pair") unless @kv == 2;
+        my ( $k, $v ) = map { _unescape_all($_) } @kv;
+        # connection string spec calls for case normalization
+        ( my $lc_k = $k ) =~ tr[A-Z][a-z];
+        if ( !$valid->{$lc_k} ) {
+            if ( $err_unsupported ) {
+                MongoDB::Error->throw("Unsupported option '$k' in URI $self\n");
```

Comment:
Here, I think we'll be less confusing if we reported the TXT records.  "... in URI $self with TXT record $txt_record ..."